src/bindings/html/dom.js

```diff
@@ -55,11 +55,14 @@ export function translateDocument(doc, langs) {
     dispatchEvent(doc, 'DOMLocalized', langs);
   };
 
-  doc.documentElement.lang = langs[0].code;
-  doc.documentElement.dir = langs[0].dir;
-  return translateFragment.call(
-    this, doc.documentElement).then(
-      setDOMLocalized, setDOMLocalized);
+  if (langs[0].code !== doc.documentElement.getAttribute('lang')) {
+    return translateFragment.call(this, doc.documentElement).then(() => {
+        doc.documentElement.lang = langs[0].code;
+        doc.documentElement.dir = langs[0].dir;
+        setDOMLocalized();
```

Comment:
Should we also call `setDOMLocalized` if `translateFragment` errors out?  The old code did that, but I'm not sure if it's the right approach.  `Promise.all` in `translateFragment` could be rejected immediately when one of the child promises rejects, which sounds like not something we'd want.